qcelemental/models/molecule.py

```diff
@@ -811,7 +852,8 @@ def from_data(
         kwarg_keys = set(kwargs.keys())
         if len(charge_spin_opts & kwarg_keys) > 0:
             for key in charge_spin_opts - kwarg_keys:
-                input_dict[key] = None
+                input_dict.pop(key, None)
+            input_dict.pop("validated", None)
```

Comment:
thinking of ways to lose data, one could `Molecule.from_data(str, masses=[something exotic])` and the masses wouldn't get considered in the __init__ validation now that we're no longer double molparse-ing here and in __init__.